roles/confluent.kafka_connect_replicator/tasks/main.yml

```diff
@@ -0,0 +1,293 @@
+  template:
+    src: kafka-connect-replicator-log4j.properties.j2
+    dest: "{{kafka_connect_replicator.log4j_file}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+  when: kafka_connect_replicator_custom_log4j|bool
+
+- name: Create Kafka Connect Replicator Jolokia Config
+  template:
+    src: kafka-connect-replicator-jolokia.properties.j2
+    dest: "{{kafka_connect_replicator_jolokia_config}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+  when: kafka_connect_replicator_jolokia_enabled|bool
+  notify: Restart Kafka Connect Replicator
+
+- name: Deploy Kafka Connect Replicator JMX Exporter Config File
+  copy:
+    src: "kafka_connect_replicator.yml"
+    dest: "{{kafka_connect_replicator_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+  when: kafka_connect_replicator_jmxexporter_enabled|bool
+
+- name: Create Kafka Connect Replicator Config
+  template:
+    src: kafka-connect-replicator.properties.j2
+    dest: "{{kafka_connect_replicator.replication_config_file}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+
+- name: Create Kafka Connect Replicator Consumer Config
+  template:
+    src: kafka-connect-replicator-consumer.properties.j2
+    dest: "{{kafka_connect_replicator.consumer_config_file}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+
+- name: Create Kafka Connect Replicator Producer Config
+  template:
+    src: kafka-connect-replicator-producer.properties.j2
+    dest: "{{kafka_connect_replicator.producer_config_file}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+
+- name: Create Kafka Connect Replicator Monitoring Interceptor Config
+  template:
+    src: kafka-connect-replicator-interceptors.properties.j2
+    dest: "{{kafka_connect_replicator.interceptors_config_file}}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+
+- name: Copy Kafka Connect Replicator Systemd script
+  template:
+    src: kafka-connect-replicator.service.j2
+    dest:  "{{kafka_connect_replicator.systemd_file}}"
+    mode: 0640
+    owner: root
+    group: root
+
+- name: Create Kafka Connect Replicator Systemd override directory
+  file:
+    path: /etc/systemd/system/kafka-connect-replicator.service.d
+    state: directory
+    mode: '0755'
+
+- name: Write Kafka Connect Replicator Service Overrides
+  template:
+    src: override.conf.j2
+    dest: "{{ kafka_connect_replicator.systemd_override }}"
+    mode: 0640
+    owner: "{{kafka_connect_replicator_user}}"
+    group: "{{kafka_connect_replicator_group}}"
+  notify: Restart Kafka Connect Replicator
```

Comment:
Does this handler get triggered?

Reply:
@domenicbove Apologies, can you clarify?

Reply:
this is in c3's tasks/main.yml
```
- name: Write Service Overrides
  template:
    src: override.conf.j2
    dest: "{{control_center.systemd_override}}"
    mode: 0640
    owner: "{{control_center_user}}"
    group: "{{control_center_group}}"
  notify: restart control center

- name: Certs were Updated - Trigger Restart
  command: /bin/true
  notify: restart control center
  when: certs_updated|bool

- meta: flush_handlers

- name: Start Control Center Service
  systemd:
    name: "{{control_center_service_name}}"
    enabled: true
    state: started

- name: Health Check
  include_tasks: health_check.yml
  when:
    - control_center_health_checks_enabled|bool
    - not ansible_check_mode
``` 

I see the certs were updated task is missing here and so is the flush handlers. flush handlers run all handlers that have been notified. and we only really have one handler in all our components, which is the restart.

its important when you run the role twice and something changes to trigger the restart. and by using the flush handler we can get the restart to happen before the health check

https://docs.ansible.com/ansible/latest/user_guide/playbooks_handlers.html#controlling-when-handlers-run